File: 1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

// @lc code=start

#include <vector>
#include <unordered_map>

using namespace std;

class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        vector<int> res;
        map<int,int> indexMap;
        auto index = 1;
        for(auto iter = nums.begin(); iter != nums.end(); iter++)
        {
            indexMap[*iter] = index++;
        }
        
       for(auto i = 0; i < nums.size(); i++) {
           if(indexMap[target-nums[i]] != 0 && (indexMap[target-nums[i]]-1) != i){
               res.push_back(i);
               res.push_back(indexMap[target-nums[i]]-1);
               return res;
           }
       }
        return res;
        
    }
};

// @lc code=end

File: 1.two-sum.java
/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> numMap = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
                numMap.put(nums[i], i);
        }
        System.out.println(numMap);
        int[] result = new int[2];
        for (int i = 0; i < nums.length; i++) {
            if (numMap.containsKey(target - nums[i]) && numMap.get(target-nums[i]) != i) {
                result[0] = i;
                result[1] = numMap.get(target - nums[i]);
                break;
            }
        }
        return result;
    }
}
// @lc code=end

